alea/include/alps/alea/plugin/stream_serializer.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 1998-2018 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#pragma once
+
+#include <numeric>
+#include <type_traits>
+
+#include <alps/alea/core.hpp>
+#include <alps/alea/complex_op.hpp>
+#include <alps/alea/mean.hpp>
+#include <alps/alea/variance.hpp>
+#include <alps/alea/covariance.hpp>
+#include <alps/alea/autocorr.hpp>
+#include <alps/alea/batch.hpp>
+
+namespace alps { namespace alea {
+
+// Metafunction to detect ALEA result types
+template<typename T> struct is_alea_result : std::false_type {};
+template<typename U> struct is_alea_result<mean_result<U>> : std::true_type {};
+template<typename U> struct is_alea_result<var_result<U>> : std::true_type {};
+template<typename U> struct is_alea_result<cov_result<U>> : std::true_type {};
+template<typename U> struct is_alea_result<autocorr_result<U>> : std::true_type {};
+template<typename U> struct is_alea_result<batch_result<U>> : std::true_type {};
+
+//
+// Serialize/deserialize complex_op<T>
+//
+
+template <typename Archive, typename T>
+void serialize(Archive &ar, complex_op<T> const &co, const unsigned int) {
+  ar & co.rere() & co.reim() & co.imre() & co.imim();
+}
+
+//
+// Serialize/deserialize *_result<T>
+//
+
+template <typename Archive> struct stream_serializer
+    : public alps::alea::serializer
+{
+
+  stream_serializer(Archive &ar) : ar_(ar) {}
+
+  void write(const std::string &, const computed<double> &data) { do_write(data); }
+  void write(const std::string &, const computed<std::complex<double>> &data) { do_write(data); };
+  void write(const std::string &, const computed<complex_op<double>> &data) { do_write(data); };
+  void write(const std::string &, const computed<long> &data) { do_write(data); };
+  void write(const std::string &, const computed<unsigned long> &data) { do_write(data); };
```

Comment:
These will have to change now, but no more allocation :-)
Also suggest `override`.